### PR TITLE
PE-4885: use new downloader on share file page

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
implement the new downloader on the shared page

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/5mdd8e2u4dmc8

### CODE REVIEW

File: lib/blocs/file_download/shared_file_download_cubit.dart [hunk 1]
```diff
@@ -28,53 +29,71 @@ class SharedFileDownloadCubit extends FileDownloadCubit {
...
-          ),
-        );
-        return;
-      }
+    String? cipher;
+    String? cipherIvTag;
+    final isPinFile = revision.pinnedDataOwnerAddress != null;
 
+    if (fileKey != null && !isPinFile) {
       final dataTx = await (_arweave.getTransactionDetails(revision.dataTxId!));
```
Comment (matibat) [inline lib/blocs/file_download/shared_file_download_cubit.dart:44]: We weren't checking if the `dataTxId` was null before... Should we?

Comment (matibat) [reply lib/blocs/file_download/shared_file_download_cubit.dart:44]: Also, the parentheses are unnecessary.

File: lib/blocs/file_download/shared_file_download_cubit.dart [hunk 2]
```diff
@@ -28,53 +29,71 @@ class SharedFileDownloadCubit extends FileDownloadCubit {
...
-          dataRes.data,
-          fileKey!,
-        );
+      if (dataTx == null) {
+        throw StateError('Data transaction not found');
       }
-    } else {
-      dataBytes = dataRes.data;
+
+      cipher = dataTx.getTag(EntityTag.cipher);
```
Comment (matibat) [inline lib/blocs/file_download/shared_file_download_cubit.dart:50]: Nit: `cipherTag` may be a more appropriate name.

File: lib/blocs/file_download/shared_file_download_cubit.dart [hunk 3]
```diff
@@ -28,53 +29,71 @@ class SharedFileDownloadCubit extends FileDownloadCubit {
...
+        return;
+      }
+
+      if (progress == 100) {
+        emit(FileDownloadFinishedWithSuccess(fileName: revision.name));
+        logger.d('Download finished');
+        return;
+      }
+
+      logger.d('Download progress: $progress');
```
Comment (matibat) [inline lib/blocs/file_download/shared_file_download_cubit.dart:81]: It's perhaps too spammy to have this log here on every progress update?

File: lib/blocs/file_download/shared_file_download_cubit.dart [hunk 4]
```diff
@@ -28,53 +29,71 @@ class SharedFileDownloadCubit extends FileDownloadCubit {
...
-        mimeType: revision.contentType ?? io.lookupMimeType(revision.name),
-        lastModified: revision.lastModifiedDate,
-      ),
+    final downloadStream = _arDriveDownloader.downloadFile(
+      dataTx: revision.dataTxId!,
+      fileName: revision.name,
+      fileSize: revision.size,
+      lastModifiedDate: revision.lastModifiedDate,
+      contentType:
+          revision.contentType ?? lookupMimeTypeWithDefaultType(revision.name),
```
Comment (matibat) [inline lib/blocs/file_download/shared_file_download_cubit.dart:60]: We're using this expression many times, let's perhaps use a final var for it.

**Review by matibat — COMMENTED** [review]
Left some comments.

File: lib/blocs/file_download/shared_file_download_cubit.dart (hunk 3 above)
Comment (thiagocarvalhodev) [reply lib/blocs/file_download/shared_file_download_cubit.dart:81]: Agreed. 045ee861989a3875eb682730bd3d45d3ec0feede

File: lib/blocs/file_download/shared_file_download_cubit.dart (hunk 1 above)
Comment (thiagocarvalhodev) [reply lib/blocs/file_download/shared_file_download_cubit.dart:44]: ba6b28d4a8ebbf96e5acd70f60dab2befd242c4b

File: lib/blocs/file_download/shared_file_download_cubit.dart (hunk 4 above)
Comment (thiagocarvalhodev) [reply lib/blocs/file_download/shared_file_download_cubit.dart:67]: 0b5a1f5278c486cd04fe2f3d4d1af3c51a492dc2

**Review by matibat — APPROVED** [review]

**Pull request merged**